python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -57,13 +60,18 @@ def dataframe(self, metric=None, mode=None):
                 rows[path].update(logdir=path)
         return pd.DataFrame(list(rows.values()))
 
-    def get_best_config(self, metric, mode="max"):
+    def get_best_config(self, metric=None, mode=None):
         """Retrieve the best config corresponding to the trial.
 
         Args:
-            metric (str): Key for trial info to order on.
-            mode (str): One of [min, max].
+            metric (str): Key for trial info to order on. Defaults to
+                ``self.default_metric``.
+            mode (str): One of [min, max]. Defaults to
+                ``self.default_mode``.
         """
+        metric = metric or self.default_metric
+        mode = mode or self.default_mode
```

Comment:
maybe define `metric = self._validate_metric(metric)` which will through an error if metric is none and self.metric is also none + other validation steps.

Reply:
same for mode